proxywasm/proxytest/http_test.go

```diff
@@ -0,0 +1,118 @@
+package proxytest
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+type testPlugin struct {
+	types.DefaultVMContext
+	buffered bool
+}
+
+type testPluginContext struct {
+	types.DefaultPluginContext
+	buffered bool
+}
+
+type testHttpContext struct {
+	types.DefaultHttpContext
+	buffered bool
+}
+
+func (p *testPlugin) NewPluginContext(uint32) types.PluginContext {
```

Comment:
Let's add doc on all experted ones

```suggestion
// NewPluginContext implements types.PluginContext NewPluginContext.
func (p *testPlugin) NewPluginContext(uint32) types.PluginContext {
```

Reply:
Even in test files? It seems a bit extreme

Reply:
yes let's do

Reply:
Got it - added docs :) Looking at what's up with the e2e